packages/teleterm/src/mainProcess/runtimeSettings.ts

```diff
@@ -79,28 +82,25 @@ function getTshHomeDir() {
   return tshPath;
 }
 
-function getTshBinaryPath() {
+// binDir is used in the packaged version to add tsh to PATH.
+// tshBinPath is used by Connect to call tsh directly.
+function getBinaryPaths(): { binDir?: string; tshBinPath: string } {
   if (app.isPackaged) {
-    return path.join(
-      getBinDir(),
+    const binDir = path.join(RESOURCES_PATH, 'bin');
+    const tshBinPath = path.join(
+      binDir,
       process.platform === 'win32' ? 'tsh.exe' : 'tsh'
     );
-  }
 
-  const tshPath = env.TELETERM_TSH_PATH;
-  if (!tshPath) {
-    throw Error('tsh path is not defined');
+    return { binDir, tshBinPath };
   }
 
-  return tshPath;
-}
-
-function getBinDir() {
-  if (!app.isPackaged) {
-    return;
+  const tshBinPath = env[TSH_BIN_ENV_VAR];
+  if (!fs.existsSync(tshBinPath)) {
+    throw Error(`${TSH_BIN_ENV_VAR} must point at a tsh binary`);
```

Comment:
In one of the upcoming commits in another PR I'll update that to `throw new Error` so that stacktrace is preserved.